assets/src/dashboard/components/toaster/provider.js

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import {
+  createContext,
+  useMemo,
+  useCallback,
+  useState,
+  useEffect,
+} from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * ToastProvider tracks messages to display based on id so that if the same message occurs in a different instance it will still display
+ * When a toast is dismissed early or at interval it is removed from the activeToasts but stays part of allToasts
+ * When useEffect clean up is called (view change) allToasts and activeToasts will reset
+ */
+export const AUTO_REMOVE_TOAST_TIME_INTERVAL = 10000;
+
+export const ToasterContext = createContext(null);
+
+const ToastProvider = ({ children }) => {
+  const [activeToasts, setActiveToasts] = useState([]);
+  const [allToasts, setAllToasts] = useState([]);
+
+  const resetToasts = useCallback(() => {
+    setAllToasts([]);
+    setActiveToasts([]);
```

Comment:
Hmm what's the need for both `activeToasts` & `allToasts` state instead of just one `toasts` state?

Reply:
So, this is going to hook into the errors seen from our stories and templates reducers. Errors  might be removed from, let's say, stories.state.error before the 10 seconds is up which we don't so much care about once we've got the error in the toaster. We also have a case where maybe that error doesn't go away from stories.state.error at the 10 seconds marker and we don't want it to reappear as a toast. I had opted to have active vs all to be a sort of gate keeper for what is a new thing to display.

that said, updating to use objects we can just use a new key in the `toasts` object to set active vs inactive and have it work just the same.

Reply:
awesome. yea I think that'll help reduce duplicate state